sonarr/releaseprofile.go

```diff
@@ -10,55 +10,103 @@ import (
 	"golift.io/starr"
 )
 
+// ReleaseProfile defines a release profile's data from Sonarr.
+type ReleaseProfile struct {
+	Enabled         bool              `json:"enabled"`
+	IncPrefOnRename bool              `json:"includePreferredWhenRenaming"`
+	ID              int64             `json:"id,omitempty"`
+	IndexerID       int64             `json:"indexerId"`
+	Name            string            `json:"name"`
+	Tags            []int             `json:"tags"`
+	Required        []string          `json:"required"`
+	Ignored         []string          `json:"ignored"`
+	Preferred       []*starr.KeyValue `json:"preferred"`
+}
+
+// Define Base Path for Release Profile calls.
+const releaseProfileBP = APIver + "/releaseProfile"
+
 // GetReleaseProfiles returns all configured release profiles.
 func (s *Sonarr) GetReleaseProfiles() ([]*ReleaseProfile, error) {
 	return s.GetReleaseProfilesContext(context.Background())
 }
 
 func (s *Sonarr) GetReleaseProfilesContext(ctx context.Context) ([]*ReleaseProfile, error) {
-	var profiles []*ReleaseProfile
+	var output []*ReleaseProfile
+
+	if _, err := s.GetInto(ctx, releaseProfileBP, nil, &output); err != nil {
+		return nil, fmt.Errorf("api.Get(releaseProfile): %w", err)
+	}
+
+	return output, nil
+}
+
+// GetReleaseProfile returns a single release profile.
+func (s *Sonarr) GetReleaseProfile(profileID int) (*ReleaseProfile, error) {
+	return s.GetReleaseProfileContext(context.Background(), profileID)
+}
+
+func (s *Sonarr) GetReleaseProfileContext(ctx context.Context, profileID int) (*ReleaseProfile, error) {
+	var output *ReleaseProfile
 
-	_, err := s.GetInto(ctx, "v3/releaseProfile", nil, &profiles)
-	if err != nil {
+	id := strconv.Itoa(profileID)
+	if _, err := s.GetInto(ctx, releaseProfileBP+"/"+id, nil, &output); err != nil {
```

Comment:
Something that occurred to me is that we can and probably should use `path.Join()` for these. What do you think about this?
```suggestion
	uri := path.Join(releaseProfileBP, strconv.Itoa(profileID))
	if _, err := s.GetInto(ctx, uri, nil, &output); err != nil {
```

Reply:
I like this, just changed!
I also changed this behaviour in the other profile related files, to keep them consistent